### PR TITLE
Added page to test Spoonacular suport

### PR DESCRIPTION
**Pull request by joeychin01**
Added two servlets, SpoonacularDishIdServlet which returns a list of recipes (and ids) from a query string, and SpoonacularRecipeServlet which returns recipe information from an id.

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -10,4 +10,24 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
\ No newline at end of file
+// limitations under the License.
+
```
Comment (ALai2) [inline src/main/webapp/script.js:14]: nit: Maybe add some comments?

**Comment by joeychin01** [comment]
Fixed merge conflicts

File: src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;  
+import com.google.gson.Gson;
+
+/** Keeping this servlet so it can be reused for formatting */
```
Comment (cocovsack) [inline src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java:29]: Update this

File: src/main/webapp/recipe.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <script src="script.js"></script>
+  </head>
+
+  <body> 
+    <input type="number" id="num-recipe" name="num-recipe" value="5738"></input>
+    <br></br>
+    <button onclick="getRecipe()">Search for recipe</button>
+    <h1 id="recipe-info"></h1>
+    <input type="text" id="dish-name" name="dish-name"></input>
+    <br></br>
```
Comment (cocovsack) [inline src/main/webapp/recipe.html:16]: Is this just for testing or is this a recipe-search functionality you are adding?

Comment (joeychin01) [reply src/main/webapp/recipe.html:16]: Just for testing, the two current servlets are being combined into one for easier use

File: src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java (hunk 2 above)
Comment (linxiz) [inline src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java:29]: Could you update the comment?

File: src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+@WebServlet("/dishId")
+public class SpoonacularDishIdServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String query = request.getParameter("dishName");
+    Client c = ClientBuilder.newClient();
```
Comment (linxiz) [inline src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java:35]: nit: avoid abbreviation for naming variables for better readability.

File: src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    WebTarget target = c.target("https://api.spoonacular.com/recipes/search?query=" + query + "&number=2&apiKey=cd2269d31cb94065ad1e73ce292374a5");
```
Comment (linxiz) [inline src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java:36]: you might want to make the prefix and the api key constants

File: src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    try {
+      String recipeInfo = target.request(MediaType.APPLICATION_JSON).get(String.class);
+      Gson gson = new Gson();
+      response.setContentType("application/json");
+      response.getWriter().println(gson.toJson(recipeInfo));
+    }
+    catch(Exception e){
```
Comment (linxiz) [inline src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java:44]: nit: one line

File: src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      System.out.println(e);
```
Comment (linxiz) [inline src/main/java/com/google/sps/servlets/SpoonacularDishIdServlet.java:45]: nit: you could use logger for server side application logging: https://cloud.google.com/appengine/docs/standard/java/logs

File: src/main/java/com/google/sps/servlets/SpoonacularRecipeServlet.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;  
+import com.google.gson.Gson;
+
+/** Keeping this servlet so it can be reused for formatting */
```
Comment (linxiz) [inline src/main/java/com/google/sps/servlets/SpoonacularRecipeServlet.java:29]: Could you update the comment?

File: src/main/java/com/google/sps/servlets/SpoonacularRecipeServlet.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+@WebServlet("/recipe")
+public class SpoonacularRecipeServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String numRecipe = request.getParameter("numRecipe");
```
Comment (linxiz) [inline src/main/java/com/google/sps/servlets/SpoonacularRecipeServlet.java:34]: Is this some kind of id or number of recipes to return?

File: src/main/webapp/script.js [hunk 10]
```diff
@@ -95,6 +96,27 @@ function clearSavedProfileStatus() {
...
+  });
+}
+
+/** Gets recipe id list from query string */
+function getRecipeId(){
+  var dishName = document.getElementById("dish-name").value;
+  fetch('/dishId?dishName='+dishName).then(response => response.json()).then((recipeId) => {
+    recipe = JSON.parse(recipeId);
+    const recipeIdDisplayElement = document.getElementById('recipe-id-info');
+    console.log(recipeId);
```
Comment (linxiz) [inline src/main/webapp/script.js:116]: is this still used?

File: src/main/java/com/google/sps/servlets/SpoonacularRecipeServlet.java (hunk 9 above)
Comment (joeychin01) [reply src/main/java/com/google/sps/servlets/SpoonacularRecipeServlet.java:34]: It is the recipe id

**Pull request merged**